### PR TITLE
feat: put user lookup behind a flag

### PR DESCRIPTION
**Pull request by hyperobject**
Adds a `!github mentions <stop/enable>` command to allow people to never get mentioned by githubbot

### CODE REVIEW

**Comment by joshblum** [comment]
perhaps this should be a per-user setting instead. users could run `!githubbot stop-mentions` and we store a flag in the db

**Comment by hyperobject** [comment]
@joshblum made the setting per-user but not per-user per-team since getting the team in the webhook handler involves a little bit hairier of a refactor than makes sense for this, I think

File: githubbot/githubbot/db.go [hunk 1]
```diff
@@ -163,3 +162,41 @@ func (d *DB) DeleteToken(identifier string) error {
...
+
+type UserPreferences struct {
+	Mention bool
+}
+
+func (d *DB) GetUserPreferences(username string) (prefs *UserPreferences, err error) {
+	row := d.DB.QueryRow(`SELECT mention
+		FROM user_prefs
+		WHERE username = ?`, username)
+	prefs = &UserPreferences{}
```
Comment (joshblum) [inline githubbot/githubbot/db.go]: naming the return var defines the variable, can nuke this line!

File: githubbot/githubbot/handler.go [hunk 2]
```diff
@@ -223,3 +228,43 @@ func (h *Handler) handleWatch(cmd string, convID string, create bool, client *gi
...
+			}
+		}
+	}()
+
+	toks, err := shellquote.Split(cmd)
+	if err != nil {
+		return fmt.Errorf(msg.ConvID, "error splitting command: %s", err)
+	}
+	args := toks[2:]
+	if len(args) != 1 || (args[0] != "stop" && args[0] != "enable") {
```
Comment (joshblum) [inline githubbot/githubbot/handler.go]: maybe `stop` -> `disable`

File: githubbot/githubbot/handler.go [hunk 3]
```diff
@@ -223,3 +228,43 @@ func (h *Handler) handleWatch(cmd string, convID string, create bool, client *gi
...
+	defer func() {
+		if message != "" {
+			_, err = h.kbc.SendMessageByConvID(msg.ConvID, message)
+			if err != nil {
+				err = fmt.Errorf("error sending message: %s", err)
+			}
+		}
+	}()
+
+	toks, err := shellquote.Split(cmd)
```
Comment (joshblum) [inline githubbot/githubbot/handler.go]: can you lowercase the command to normalize

File: githubbot/githubbot/handler.go [hunk 4]
```diff
@@ -223,3 +228,43 @@ func (h *Handler) handleWatch(cmd string, convID string, create bool, client *gi
...
+		return fmt.Errorf(msg.ConvID, "error splitting command: %s", err)
+	}
+	args := toks[2:]
+	if len(args) != 1 || (args[0] != "stop" && args[0] != "enable") {
+		message = "I don't understand! Try `!github mentions stop` or `!github mentions enable`."
+		return nil
+	}
+
+	if args[0] == "enable" {
+		err = h.db.SetUserPreferences(msg.Sender.Username, &UserPreferences{Mention: true})
```
Comment (joshblum) [inline githubbot/githubbot/handler.go]: cleaner:
```
allowMentions = args[0] == "enable"
err = h.db.SetUserPreferences(msg.Sender.Username, &UserPreferences{Mention: allowMentions})
```
and then toggle the message based on `allowMentions`

File: githubbot/githubbot/db.go (hunk 1 above)
Comment (hyperobject) [reply githubbot/githubbot/db.go]: @joshblum was getting nil pointer errors without this line, not sure how to resolve that

**Pull request merged**